snap/snapcraft.yaml

```diff
@@ -987,3 +989,12 @@ parts:
       juju: bin/juju
     prime:
       - bin/juju
+
+  nano:
```

Comment:
It _could_ be, my thinking was A) it's not a "bash utility" and B) it seems more correct to break the parts down this way.

Reply:
But should we have a separate part for each binary we bring from the the deb repos? That looks overly verbose. Let's keep the current approach of pulling everything we need from the deb repos in a single part and rename the "bash-utils" to something denoting that are not only bash utils.

Reply:
It is verbose, but I believe this is the correct choice.  If we end up needing to add more permissive interfaces to just a since "bash-util" (nano is not a bash util), then every bash util will also inherit this, which may be undesirable and will make justifying the autoconnection of these, to the snapstore team, more difficult.

Reply:
It is verbose, but I believe this is the correct choice. If we end up needing to add more permissive interfaces to just a since "bash-util" (nano is not a bash util), then every bash util will also inherit this, which may be undesirable and will make justifying the autoconnection of these, to the snapstore team, more difficult.

Reply:
I agree, If there are interfaces we need to connect then this is the correct choice. Do we currently have any interfaces we want to connect?

Reply:
Also this nano part pulls into the snap only the nano binary and not any of its library dependencies. Maybe this time we are lucky and the dependencies are in the core18 base. Do we know if this will be the case in the future, what if core20 does not have the same libraries included.

This approach of splitting all the packages we bring from the deb repository into separate parts and cherry picking the files we believe are needed worries me. I am not confident I can keep up with the changes that may happen in the distro.

If the "bash-utils" name is the problem lets rename that part to something more generic eg "base-repo-packages".

As soon as we find ourselves in need of connecting interfaces to specific binaries from within the "base-repo-packages" that would be the time where we should consider splitting some packages out of it.

Reply:
> If we end up needing to add more permissive interfaces to just a since "bash-util" (nano is not a bash util), then every bash util will also inherit this, which may be undesirable and will make justifying the autoconnection of these, to the snapstore team, more difficult.

I am not sure I follow what you are saying here. The plugs are related to the snap commands and not the snap parts. What does it matter which part nano comes from?